raphtory/src/search/mod.rs

```diff
@@ -1,1498 +1,367 @@
-            .collect::<Vec<_>>();
-        let expected = vec![("Frodo".to_string(), "Gollum".to_string())];
-        assert_eq!(actual, expected);
-    }
-
-    #[test]
-    fn search_by_edge_props_graph_indexed() {
-        let g = Graph::new();
-        g.add_edge(
-            1,
-            "Frodo",
-            "Gandalf",
-            [("type", Prop::str("friends"))],
-            None,
-        )
-        .expect("add edge failed");
-        g.add_edge(1, "Frodo", "Gollum", [("type", Prop::str("enemies"))], None)
-            .expect("add edge failed");
-
-        let g: IndexedGraph<Graph> = g.into();
-
-        let results = g
-            .search_edges(r#"type:'friends'"#, 10, 0)
-            .expect("search failed");
-        let actual = results
-            .into_iter()
-            .map(|e| (e.src().name(), e.dst().name()))
-            .collect::<Vec<_>>();
-        let expected = vec![("Frodo".to_string(), "Gandalf".to_string())];
-
-        assert_eq!(actual, expected);
-
-        let results = g
-            .search_edges(r#"type:'enemies'"#, 10, 0)
-            .expect("search failed");
-        let actual = results
-            .into_iter()
-            .map(|e| (e.src().name(), e.dst().name()))
-            .collect::<Vec<_>>();
-        let expected = vec![("Frodo".to_string(), "Gollum".to_string())];
-
-        assert_eq!(actual, expected);
-    }
-
-    #[test]
-    fn search_by_edge_src_dst() {
-        let g = Graph::new();
-        g.add_edge(1, "Frodo", "Gandalf", NO_PROPS, None)
-            .expect("add edge failed");
-        g.add_edge(1, "Frodo", "Gollum", NO_PROPS, None)
-            .expect("add edge failed");
-
-        let ig: IndexedGraph<Graph> = g.into();
-
-        let results = ig
-            .search_edges(r#"from:Frodo"#, 10, 0)
-            .expect("search failed");
-        let mut actual = results
-            .into_iter()
-            .map(|e| (e.src().name(), e.dst().name()))
-            .collect::<Vec<_>>();
-        let mut expected = vec![
-            ("Frodo".to_string(), "Gandalf".to_string()),
-            ("Frodo".to_string(), "Gollum".to_string()),
-        ];
-
-        actual.sort();
-        expected.sort();
-
-        assert_eq!(actual, expected);
-
-        // search by destination
-        let results = ig.search_edges("to:gollum", 10, 0).expect("search failed");
-        let actual = results
-            .into_iter()
-            .map(|e| (e.src().name(), e.dst().name()))
-            .collect::<Vec<_>>();
-        let expected = vec![("Frodo".to_string(), "Gollum".to_string())];
-
-        assert_eq!(actual, expected);
+fn index_node_temporal_properties<
+    'g,
+    PI: DerefMut<Target = Vec<Option<PropertyIndex>>>,
+    G: GraphViewOps<'g>,
+    GH: GraphViewOps<'g>,
+>(
+    node: NodeView<G, GH>,
+    properties: Vec<(i64, ArcStr, usize, Prop)>,
+    mut property_indexes: PI,
+    node_id: u64,
+    writers: &[Option<IndexWriter>],
+) -> tantivy::Result<()> {
```

Comment:
This is very strange to me, we seem to be indexing everything in a double for loop and checking the values equal values. When we call this we know what properties are being added and we changed the index structure so we don't index everything on every timestamp.

This happens for edges as well